src/app/fdctl/config.c

```diff
@@ -759,7 +759,12 @@ config_parse( int *    pargc,
   } else {
     if( FD_UNLIKELY( !if_nametoindex( result.tiles.net.interface ) ) )
       FD_LOG_ERR(( "configuration specifies network interface `%s` which does not exist", result.tiles.net.interface ));
-    result.tiles.net.ip_addr = listen_address( result.tiles.net.interface );
+      
+    uint iface_ip = listen_address( result.tiles.net.interface );
+    if ( FD_UNLIKELY( !fd_ip4_addr_is_public ( iface_ip ) ) )
```

Comment:
I think we probably want to allow this for non-public clusters. Eg, if `result.is_live_cluster` below, then it must be public but otherwise private subnets are ok.